crates/proof/src/preloaded_eigenda_provider.rs

```diff
@@ -30,15 +30,16 @@ impl From<EigenDABlobWitnessData> for PreloadedEigenDABlobProvider {
         let mut entries = vec![];
 
         for i in 0..blobs.len() {
-            // always verify steel proof if in zkvm mode
-            assert!(validate_validity_proof(
+            // always verify validity of the cert
+            value.validity[i].validate_cert_receipt(
                 &value.eigenda_certs[i],
-                &value.validity_proofs[i]
-            ));
+                // TODO figure out a way to pass down validity_call_verifier_id
+                // at minimum, this value needs to come from system config from derivation
+                B256::default(),
+            );
 
             // if valid, check blob kzg integrity
-            let is_valid = value.validity_proofs[i].0;
-            if is_valid {
+            if value.validity[i].claimed_validity {
```

Comment:
think we can get rid of the CertValidity struct if `validate_cert_receipt` returns `validity` (which wouldn't be a claimed validity anymore, it would be the actual validity).